paddle/fluid/framework/init.cc

```diff
@@ -62,6 +62,18 @@ void InitDevices() {
 
   for (int i = 0; i < count; ++i) {
     places.emplace_back(platform::CUDAPlace(i));
+    for (int j = 0; j < count; ++j) {
+      if (i == j) continue;
+      int can_acess = -1;
+      PADDLE_ENFORCE(cudaDeviceCanAccessPeer(&can_acess, i, j),
+                     "Failed to test P2P access.");
+      if (can_acess != 1) {
+        VLOG(1) << "Cannot enable P2P access from " << i << " to " << j;
+      } else {
+        cudaSetDevice(i);
+        cudaDeviceEnablePeerAccess(j, 0);
```

Comment:
Is P2P Access need any specific hardware? I vaguely remember that it needs the slim-connect(or something)?

Reply:
It needs to be tesla. So gforce titan x doesn't support it